ONSdigital/response-operations-ui
# Added button to allow generation of IAC

The designs for this are still under review so this is functionally
complete but look and feel left deliberately unfinished.  Also tests
to be added

# Motivation and Context
For the social pilot, SEL need to be able to generate a new UAC for a respondent who phones up.  This change adds a button to do that, leading to a new IAC being displayed on the 'view case details' page until it is refreshed.  The functionality for this is complete but tests need to be added (which is why currently showing wip label)

# What has changed
- added interactions with the CaseIACEndpoint in the case service to the case controller
- factored out construction of the model for the 'view case details' page to avoid code duplication between default viewing and IAC generation
- added a 'Generate UAC' button and a message indicating the number of IACs that have been generated for a case to the 'view case details' page

# How to test?
- make down pull up
- check out this branch of response-operations-ui and build/deploy to docker
- check out the [generate_new_access_code branch of rm-case-service](https://github.com/ONSdigital/rm-case-service/tree/generate_new_access_code)  and build/deploy to docker
- navigate to the 'view case details' page of the social area of response-operations-ui
- confirm the count of existing iacs is correct
- generate a new iac
- confirm the iac works in RH
- confirm the count is incremented on the 'view case details' page

# Links
[generate_new_access_code branch of rm-case-service](https://github.com/ONSdigital/rm-case-service/tree/generate_new_access_code)
